### PR TITLE
replace --forward-port directions with --add-port in the port-forwarding page

### PR DESCRIPTION
**Pull request by yyolk**
- replacing `--forward-port` directions with `--add-port`
  - readability add note for NAT and routing (thanks @mDuo13 !)
- add note for virtual firewall and attaching 

I'm new to firewalld, but I think the purpose of the [`--add-forward-port`](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-port_forwarding) is if it's served internally on a different port. 

Update: removed a bunch of cruft around being a bit of a novice in running a node. I've now successfully done it on bare metal (requiring a firewall), and on various VPS several times including.

I've run a test node on 'bare metal' for a month, which I opted in using `firewalld` to the internet.

I started a fresh one 5 days ago using the latest commit's [changes as detailed in this post](https://github.com/ripple/xrpl-dev-portal/pull/1105#issuecomment-873455189) (caught my wrong long option :sweat_smile:)

### CODE REVIEW

**Review by SirJosh1987 — APPROVED** [review]

**Review by SirJosh1987 — APPROVED** [review]

**Review by SirJosh1987 — APPROVED** [review]
Am I on the right path so far

File: content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md [hunk 1]
```diff
@@ -35,10 +35,24 @@ Loading: "/etc/opt/ripple/rippled.cfg"
...
+To allow incoming connections, configure your firewall to forward the peer protocol port, which is served on **port 51235** in the default config file. The instructions to forward a port depend on your firewall. For example, if you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-using_zones_to_manage_incoming_traffic_depending_on_source) to open **port 51235** to all incoming traffic.
+
+_Assuming `--zone=public` is your public [zone](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-working_with_zones#sec-Listing_Zones)._
+
+```sh
+$ sudo firewall-cmd --zone=public --add-source-port=51235/tcp
+```
+
+It's a good idea to restart rippled to see it's effects:
```
Comment (mDuo13) [inline content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md]:
```suggestion
Then, restart the `rippled` server:
```

File: content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md [hunk 2]
```diff
@@ -35,10 +35,24 @@ Loading: "/etc/opt/ripple/rippled.cfg"
 }
 ```
 
-To allow incoming connections, configure your firewall to forward the peer protocol port, which is served on **port 51235** in the default config file. The instructions to forward a port depend on your firewall. For example, if you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-port_forwarding) to forward TCP traffic as follows:
+To allow incoming connections, configure your firewall to forward the peer protocol port, which is served on **port 51235** in the default config file. The instructions to forward a port depend on your firewall. For example, if you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-using_zones_to_manage_incoming_traffic_depending_on_source) to open **port 51235** to all incoming traffic.
```
Comment (mDuo13) [inline content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md]:
```suggestion
To allow incoming connections, configure your firewall to allow incoming traffic on the peer protocol port, which is served on **port 51235** in the default config file. The instructions to open a port depend on your firewall. If your server is behind a router that performs Network Address Translation (NAT), you must configure your router to forward the port to your server.

If you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-using_zones_to_manage_incoming_traffic_depending_on_source) to open **port 51235** to all incoming traffic.
```

**Review by mDuo13 — CHANGES_REQUESTED** [review]
- Rephrased the "restart" bit. (You had the wrong "its" there)
- While we're here, a misconception I've seen a couple times is that people think they're not behind a firewall even though their NAT router is acting as a firewall. So I suggested a rephrased first paragraph to clarify that.
- I'm not familiar enough with `firewalld` to confirm that your instructions are right, but I think they might be. I'll see if I can get someone else with more expertise to double-check.

File: content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md (hunk 2 above)
Comment (yyolk) [reply content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md]: Great suggestion!

**Comment by yyolk** [comment]
>     * Rephrased the "restart" bit. (You had the wrong "its" there)
Thank you :)

>     * While we're here, a misconception I've seen a couple times is that people think they're not behind a firewall even though their NAT router is acting as a firewall. So I suggested a rephrased first paragraph to clarify that.

Totally makes sense! I encountered this page looking for _"the port to forward"_ having the NAT bit is useful context. :+1: 

 
>     * I'm not familiar enough with `firewalld` to confirm that your instructions are right, but I think they might be. I'll see if I can get someone else with more expertise to double-check.

I believe this as well. I'm far from an expert with `firewalld`, I did try the previous `--add-forward-port` on a bare metal server with `firewalld` and didn't get anywhere. I'll run through this again to confirm.

**Comment by yyolk** [comment]
here's documentation running through ~~`--add-forward-port`~~ `--add-port` (*[see checklist item](#lolwhoops) where i added commit https://github.com/ripple/xrpl-dev-portal/pull/1105/commits/54e6ea2866220245af196a5ff2d81a6ea4d2201a*)

starting from a new instance, no rules are for forward or open ports:

image

adding the forward-port
image


initial sync, seems that more peers are being seen that anticipated (i recall seeing 10 here, but this was before i started to set `[peers_max]`)
image

still not "FULL", this node is connecting to peers but nothing is able to connect to it:

image

node is "FULL", no peers are connecting to it:
- image
- image


* * *
stopped `rippled` and removed forward rule and add open port rule

image

* * *

starting up and waiting no time, there are no peers:

image


so i deleted the peerfinder sqllite db in `/var/lib/rippled/db/`

***turns out i actually need to use `--add-port` (which matches my production-test node, not this new test) :sweat_smile:***

image


- [X] <a name="lolwhoops"></a> fix to use `--add-port`, not `--add-source-port`


immediately i see nodes connect to me with it
image
image

finally an established connection:
image

waited a little bit, now i'm seeing peers on the node list
image

**Comment by yyolk** [comment]
added a note around virtual firewalls, updated my pull description

File: content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md [hunk 3]
```diff
@@ -35,14 +35,32 @@ Loading: "/etc/opt/ripple/rippled.cfg"
...
+
+To make it permanent:
+
+```sh
+$ sudo firewall-cmd --zone=public --permanent --add-port=51235/tcp
 ```
 
 For other software and hardware firewalls, see the manufacturer's official documentation.
 
+If your using a hosting provider includes a virtual firewall, you do not need to use `firewalld`, but will still need to allow the peer port, and ensure your host is attached. e.g., AWS Security Groups with a rule for the peer port open to the public internet.
```
Comment (mDuo13) [inline content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md]:
```suggestion
If you are using a hosting service with a virtual firewall (for example, [AWS Security Groups](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html)), you do not need to use `firewalld`, but you still need to allow inbound traffic from the open internet on the peer port. Make sure you apply the relevant rules to your host or virtual machine.
```

**Review by mDuo13 — CHANGES_REQUESTED** [review]
Looks like "your hosting provider" and "you're using a hosting provider" got mixed up into a typo here. I suggested a rephrased version to clarify. (Not sure what "ensure your host is attached" is supposed to have meant there—I took a guess that you mean to remember to attach the rules to the relevant machine.)

**Comment by SirJosh1987** [comment]
I'm slowly learning

File: content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md (hunk 2 above)
Comment (SirJosh1987) [reply content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md]: How?

**Comment by yyolk** [comment]
Thank you!

**Pull request merged**